x/gov/msgs.go

```diff
@@ -43,31 +44,11 @@ func (msg MsgSubmitProposal) Type() string  { return TypeMsgSubmitProposal }
 
 // Implements Msg.
 func (msg MsgSubmitProposal) ValidateBasic() sdk.Error {
-	if len(msg.Title) == 0 {
-		return ErrInvalidTitle(DefaultCodespace, "No title present in proposal")
-	}
-	if len(msg.Title) > MaxTitleLength {
-		return ErrInvalidTitle(DefaultCodespace, fmt.Sprintf("Proposal title is longer than max length of %d", MaxTitleLength))
-	}
-	if len(msg.Description) == 0 {
-		return ErrInvalidDescription(DefaultCodespace, "No description present in proposal")
-	}
-	if len(msg.Description) > MaxDescriptionLength {
-		return ErrInvalidDescription(DefaultCodespace, fmt.Sprintf("Proposal description is longer than max length of %d", MaxDescriptionLength))
+	ty := msg.ProposalType
+	if ty != ProposalTypeText && ty != ProposalTypeSoftwareUpgrade {
```

Comment:
```suggestion
	if ty != ProposalTypeText || ty != ProposalTypeSoftwareUpgrade {
```

Reply:
If then isn't the conditional evaluated `true` even when `ty == ProposalTypeText`, because of `ty != ProposalTypeSoftwareUpgrade`, which leads to an error? (also true for the case of `ty == ProposalTypeSoftwareUpgrade` which is detected by `ty != ProposalTypeText`) 

The new statement always evaluated `true` unless `ProposalTypeText == ProposalTypeSoftwareUpgrade == ty`

Reply:
I can change it to `!(ty == ProposalTypeText || ty == ProposalTypeSoftwareUpgrade)` which seems easier to read

Reply:
I think I see what you mean. Still, the conditional reads weird. What is the primitive type of `ProposalType`? What it really should be is something like: `if !validTypes.Includes(ty) { // error}`. Especially as the set of valid types increases over time.

Reply:
> I can change it to `!(ty == ProposalTypeText || ty == ProposalTypeSoftwareUpgrade)` which seems easier to read

Yes, much easier to read! But see above comment.

Reply:
`ProposalType` is just a string. The restriction is `gov` module specific which accepts two kinds `TextProposal` and `SoftwareUpgradeProposal`. It is ideal to have separate message type for each proposal type but it will break the api(even more). So the other proposal submitting msg types will have single conditional expression(like as in paramchange proposal)

Do you think it is better to unify the format into either msg-per-parameter or msg-per-module? If so I can change

Reply:
Good question. We should discuss further, but it seems message-per-module seems more flexible, no?